pkg/volume/csi/csi_mounter.go

```diff
@@ -300,7 +300,6 @@ func (c *csiMountMgr) TearDownAt(dir string) error {
 
 	if !mounted {
 		klog.V(4).Info(log("unmounter.Teardown skipping unmount, dir not mounted [%s]", dir))
-		return nil
```

Comment:
When the dir is already unmounted (or was never mounted), the volume controller will call `tearDown` and ask he backing driver to do a NodeUnpublish on a volume that may already have been unpublished.  This could cause an error to be returned by a driver implementation that is unable to unpublish and already unpublished volume.

So, continuing beyond this point when  dir is not mounted maybe problematic.  Maybe it should be something like this

```go
if !mounted {
    klog(...)
    // try to remove dir 
    return removeDir(c.plugin, dir)
}
```
That way, if `mounted == true`, there is always an attempt to do NodeUnpublish.

Reply:
@vladimirvivien  Address the comments. PTAL, thanks